### DISCUSSION TITLE
为什么plugin-tab不能转换以非tabs语法开头的md内容

### DISCUSSION THREAD

**Discussion by weixq709** [mdit-plugins/mdit-plugins#44] [Q&A]
**插件支持的转换方式**
::: tabs#fruit

@tab apple#apple

Apple

@tab banana#banana

Banana

:::

**添加其他内容**
test
::: tabs#fruit

@tab apple#apple

Apple

@tab banana#banana

Banana

:::
通常我们会在md中包含很多元素，比如：标题、文本、代码块等等，如果使用上面的这种方式，不是只能在md中包含选项卡语法，无法包含其他md语法？

**Comment by Mister-Hope** [comment]
我完全没听懂你在说什么，我建议你用代码块来准确的给出示例代码，然后说出你觉得哪里不工作

**Reply by weixq709** [reply]
import markdownIt from 'markdown-it'
import { tab } from '@mdit/plugin-tab'

const md = markdownIt();
md.use(tab);

const input = `
<\p>文本内容<\/p>
::: tabs#fruit
@tab apple#apple
Apple

@tab banana#banana
Banana
:::`;

const output = newMd.render(input);
console.log(output);

输出结果
\<p>文本内容<\/p>
\<pre\>\<code\>  ::: tabs#fruit
\<\/code\>\<\/pre\>
\<p\>@tab apple#apple\<\/p\>
\<p\>Apple\<\/p\>
\<p\>@tab banana#banana\<\/p\>
\<p\>Banana\<\/p\>
\<p\>:::\<\/p\>

期望输出
\<p\>文本内容\<\/p\>
\<div class="tabs-tabs-wrapper" data-id="fruit"\>
  \<div class="tabs-tabs-header"\>
    \<button type="button" class="tabs-tab-button" data-tab="0" data-id="apple"\>apple\<\/button\>
    \<button type="button" class="tabs-tab-button" data-tab="1" data-id="banana"\>banana\</button\>
  \<\/div\>
  \<div class="tabs-tabs-container"\>
    \<p\>Apple\<\/p\>
   \<p\>Banana\<\/p\>
 \<\/div\>
\<\/div\>

**Reply by Mister-Hope** [reply]
What if you make sure the tab is surrounded by empty line?

**Reply by Mister-Hope** [reply]
Closing this one. you are supposed to learn more markdowns basics,  Such as how does markdown treat line break

**Reply by weixq709** [reply]
谢谢，我解决了
